src/hooks/useRefState.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * This hook allows to use a Ref similar to how `useState`
+ * operates to store data. Essentially you initialize it
+ * with an initialValue and it returns the value of the ref
+ * + a function to set the state.
+ *
+ * An example of where this is used would be in event listeners
+ * (window.addEventListener) where normal state can not be accessed.
+ */
+export const useRefState = (initialValue: any = null) => {
+  const state = useRef(initialValue);
+
+  // Cleanup the state
+  useEffect(() => {
+    return () => {
+      state.current = null;
+    };
+  }, []);
+
+  const getState = () => {
+    return state.current;
+  };
+
+  const setState = (fn: any, dontCopy = false) => {
```

Comment:
ended up adding this 2nd param to handle not copying the fns & authClient. I didn't want to change the default behavior since it would be confusing to have the fn behave differently in feathery-react and feathery-frontend

I guess ideally this would be exported and used in feathery-frontend